src/praisonai-agents/praisonaiagents/config/feature_configs.py

```
@@ -732,9 +732,10 @@ class ExecutionConfig:
     rate_limiter: Optional[Any] = None
 
     # Context compaction: automatically compact chat_history when approaching token limit.
-    # Opt-in only (safe default: False). Uses existing praisonaiagents.compaction module.
-    # Usage: Agent(execution=ExecutionConfig(context_compaction=True, max_context_tokens=8000))
-    context_compaction: bool = False
+    # Now ENABLED by default for proactive overflow protection (opt-out with False).
+    # Usage: Agent(execution=ExecutionConfig(context_compaction=False))  # to disable
+    # Or: Agent(execution=ExecutionConfig(context_compaction=my_policy))  # custom policy
+    context_compaction: Union[bool, "ContextCompactionPolicy"] = True
```

Ruff (0.15.15)

[error] Undefined name `ContextCompactionPolicy`

(F821)